src/python/pants/backend/codegen/tasks/simple_codegen_task.py

```diff
@@ -172,3 +217,101 @@ def execute(self):
 
       if self.artifact_cache_writes_enabled():
         self.update_artifact_cache(vts_artifactfiles_pairs)
+
+
+  class CodegenStrategy(object):
```

Comment:
Good question.  Let's leave it embedded in SimpleCodegenTask for now.  I am concerned that if we were to try to extend this concept to the logic in CodeGen it would get cluttered with extra arguments and more complex return values to handle the different language issues.